source/server/http/admin.cc

```diff
@@ -192,7 +192,11 @@ Http::FilterHeadersStatus AdminFilter::decodeHeaders(Http::HeaderMap& headers, b
   return Http::FilterHeadersStatus::StopIteration;
 }
 
-Http::FilterDataStatus AdminFilter::decodeData(Buffer::Instance&, bool end_stream) {
+Http::FilterDataStatus AdminFilter::decodeData(Buffer::Instance& data, bool end_stream) {
+  // Currently we generically buffer all admin request data in case a handler wants to use it.
+  // If we ever support streaming admin requests we may need to revisit this.
+  callbacks_->addDecodedData(data, false);
```

Comment:
as I read this, we're doing addDecodedData(..., false) and StopIterationNoBuffer rather than returning StopIterationAndBuffer because we need the complete body to be available onComplete, and there's really no other way to make that happen, yeah?   If so, I'd like at least a comment and at best a unit test that'd catch if we did it wrong :-)

Reply:
Yeah that's right. I will add more comments. There is a UT I added. Also, this path is covered in an integration test in the other main tap PR.